src/Compilers/CSharp/Portable/Lowering/InitializerRewriter.cs

```diff
@@ -65,7 +65,7 @@ internal static BoundTypeOrInstanceInitializers RewriteScriptInitializer(Immutab
                 Debug.Assert(!submissionResultType.IsVoidType());
 
                 // Note: The trailing expression was already converted to the submission result type in Binder.BindGlobalStatement.
-                boundStatements.Add(new BoundReturnStatement(lastStatement.Syntax, RefKind.None, trailingExpression));
+                boundStatements.Add(new BoundReturnStatement(lastStatement.Syntax, RefKind.None, trailingExpression, @checked: false));
```

Comment:
> What if the last expression was a checked expression? Or would that be fine, because the return itself isn't checked, but the expression it contains is?

This flag is not interesting post lowering. It is only interesting for inference for lambdas and is supposed to reflect the context to use for a possible conversion if return statement is supposed to perform one. The actual expression in this case makes no impact on the context for conversion.

Reply:
I'll add a comment to BoundNodes.xml